pkg/cmd/component/init.go

```diff
@@ -0,0 +1,333 @@
+			if err != nil {
+				log.Progressf("failed to create the directory %s with application name %s : %v", io.ApplicationFolder, appName[len(appName)-1], err)
+				io.ApplicationFolder = ui.ApplicationOutputPath()
+			}
+		} else {
+			listDir := ListFiles(appFS, io.ApplicationFolder)
+			if len(listDir) != 0 {
+				exists, _ = ioutils.IsExisting(appFS, filepath.Join(io.ApplicationFolder, "components"))
+				if !exists {
+					log.Progressf("the given Path : %s is not a correct path for an application ", io.ApplicationFolder)
+					io.ApplicationFolder = ui.ApplicationOutputPath()
+				}
+			}
+		}
+
+	}
+	if io.ApplicationFolder == "" {
+		io.ApplicationFolder = ui.ApplicationOutputPath()
+	}
+	// ask for confirmation before creating the directory
+	exists, _ := ioutils.IsExisting(appFS, io.ApplicationFolder)
+	if !exists {
+		check := ui.AskConfirmation(io.ApplicationFolder)
+		if check {
+			appFS.MkdirAll(io.ApplicationFolder, 0755)
+		} else {
+			return fmt.Errorf("interactive mode has been terminated")
+		}
+	}
+
+	if !checkGit(appFS, io.ApplicationFolder) {
+		if io.GitRepoURL == "" {
+			io.GitRepoURL = ui.EnterGitRepoURL()
+		}
+		url, err = url.Parse(io.GitRepoURL)
+		for err != nil {
+			log.Progressf("failed to parse GitOps repo URL %q: %v", io.GitRepoURL, err)
+			io.GitRepoURL = ui.EnterGitRepoURL()
+			url, err = url.Parse(io.GitRepoURL)
+		}
+		io.GitRepoURL = utility.AddGitSuffixIfNecessary(io.GitRepoURL)
+		if !isKnownDriverURL(io.GitRepoURL) {
+			io.PrivateRepoURLDriver = ui.SelectPrivateRepoDriver()
+			host, err := accesstoken.HostFromURL(io.GitRepoURL)
+			if err != nil {
+				return fmt.Errorf("failed to parse the gitops url: %v", err)
+			}
+			identifier := factory.NewDriverIdentifier(factory.Mapping(host, io.PrivateRepoURLDriver))
+			factory.DefaultIdentifier = identifier
+		}
+		secret, err := accesstoken.GetAccessToken(io.GitRepoURL)
+		if err != nil && err != keyring.ErrNotFound {
+			return err
+		}
+		if secret == "" {
+			// We must prompt for the token
+			if io.Secret == "" {
+				io.Secret = ui.EnterGitSecret(io.GitRepoURL)
+				io.SaveTokenKeyRing = ui.UseKeyringRingSvc()
+				if io.SaveTokenKeyRing {
+					setAccessTokenInit(io)
+				}
+			}
+		}
+		io.Secret = secret
+	} else {
+		return fmt.Errorf("git repo already been initiated")
+	}
+	return gitInitializeCheck(io)
+}
+
+// Checks the requirement for .git folder
+func gitInitializeCheck(io *InitParameters) error {
+	u, _ := url.Parse(io.GitRepoURL)
+	parts := strings.Split(u.Path, "/")
+	org := parts[1]
+	repoName := strings.TrimSuffix(strings.Join(parts[2:], "/"), ".git")
+	u.User = url.UserPassword("", io.Secret)
+	client, err := factory.FromRepoURL(u.String())
+	if err != nil {
+		return fmt.Errorf("failed to create a client to access %q: %v", io.GitRepoURL, err)
+	}
+	ctx := context.Background()
+	currentUser, _, err := client.Users.Find(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get the user with their auth token: %v", err)
```

Comment:
When I tried the init command, I'm getting this error.  Are you getting it?

Reply:
No I am not getting this error.